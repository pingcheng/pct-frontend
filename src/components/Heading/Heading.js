import React, { Component } from "react";
import PropTypes from "prop-types";

/**
 * Heading component.
 */
export class Heading extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
      PropTypes.object,
    ]),
    align: PropTypes.string,
    level: PropTypes.number,
    className: PropTypes.string,
  };

  static defaultProps = {
    subTitle: null,
    align: "left",
    level: 1,
    className: "",
  };

  render() {
    let subTitle = "";

    if (this.props.subTitle) {
      subTitle = (
        <div className="text-sm text-gray-500">{this.props.subTitle}</div>
      );
    }

    return (
      <div>
        <h1
          className={`mb-8 ${this.props.className}`}
          style={{
            textAlign: this.props.align,
          }}
        >
          <div>{this.props.title}</div>
          {subTitle}
        </h1>
      </div>
    );
  }
}
