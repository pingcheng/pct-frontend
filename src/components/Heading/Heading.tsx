export type HeadingProps = {
  title: string;
  subTitle?: string;
  align?: "left" | "right" | "center";
  className?: string;
};

/**
 * Heading component.
 */
export function Heading({
  title,
  subTitle,
  align = "left",
  className,
}: HeadingProps): JSX.Element {
  return (
    <div>
      <h1
        className={`mb-8 ${className ?? ""}`}
        style={{
          textAlign: align,
        }}
        role="heading"
      >
        <div>{title}</div>
        {subTitle && <div className="text-sm text-gray-500">{subTitle}</div>}
      </h1>
    </div>
  );
}
