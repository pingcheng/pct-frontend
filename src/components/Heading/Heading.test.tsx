import { render, screen } from "@testing-library/react";
import { Heading } from "./Heading";

describe("test <Heading />", () => {
  const props = {
    title: "Title",
    subTitle: "Sub-title",
  };

  it("should render correctly", () => {
    const { container } = render(<Heading {...props} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("should render the title text", () => {
    render(<Heading {...props} />);
    expect(screen.getByText(props.title)).toBeInTheDocument();
  });

  it("should render the sub title text", () => {
    render(<Heading {...props} />);
    expect(screen.getByText(props.subTitle)).toBeInTheDocument();
  });

  it("should apply align when specific align mode", () => {
    render(<Heading {...props} align="right" />);
    expect(screen.getByRole("heading")).toHaveStyle({
      textAlign: "right",
    });
  });

  it("should apply extra class name when specific class name", () => {
    render(<Heading {...props} className="extra-class-name" />);
    expect(screen.getByRole("heading")).toHaveClass("extra-class-name");
  });
});
