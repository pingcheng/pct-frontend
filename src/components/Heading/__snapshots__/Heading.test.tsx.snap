// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Heading /> should render correctly 1`] = `"<div><h1 class=\\"mb-8 \\" style=\\"text-align: left;\\" role=\\"heading\\"><div>Title</div><div class=\\"text-sm text-gray-500\\">Sub-title</div></h1></div>"`;
